File: tests/test_php/src/Services/UserService.php
<?php

namespace App\Services;

use App\Models\User;
use Monolog\Logger;

class UserService {
    private $logger;

    public function __construct() {
        $this->logger = new Logger('UserService');
    }

    public function save(User $user) {
        $this->logger->info("Saving user: " . $user->getName());
    }

    public function findAll() {
        return array();
    }

    public function update(User $user) {
        $this->logger->info("Updating user: " . $user->getName());
        return true;
    }
}
?>

File: tests/test_lua/services/user_service.lua
local User = require("models.user")

local UserService = {}
UserService.__index = UserService

function UserService.new()
    local self = setmetatable({}, UserService)
    self.users = {}
    return self
end

function UserService:save(user)
    table.insert(self.users, user)
    print("Saved user: " .. user:getName())
end

function UserService:findAll()
    return self.users
end

return UserService
